wormhole/file_transport.go

```diff
@@ -382,11 +415,19 @@ func (t *fileTransport) makeTransitMsg() (*transitMsg, error) {
 			return nil, fmt.Errorf("port isn't an integer? %s", portStr)
 		}
 
+		var relayType string;
+		if t.relayProto == "tcp" {
+			relayType = "direct-tcp-v1"
+		} else if t.relayProto == "ws" {
+			relayType = "direct-ws-v1"
+		} else {
+			panic("unknown relay protocol")
```

Comment:
We should return a nil and and error rather than panic: `return nil, fmt.Errorf("unknown relay protocol")`. Since this function already returns an error we can take advantage of that. Panic isn't intended to be used for exception handling, to respond to this panic there would have to be a `defer` call somewhere higher in the stack which calls `recover()`.

See [the golang blog post](https://blog.golang.org/defer-panic-and-recover)